### PR TITLE
python3Packages.textblob: init at 0.18.0

### PR DESCRIPTION
**Pull request by idlip**
## Description of changes

Home: https://textblob.readthedocs.io/

Simple, Pythonic text processing. Sentiment analysis, part-of-speech tagging, noun phrase parsing, and more.

Its a text mining that goes hand in hand with common natural language processing by using `python3Packages.nltk`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/textblob/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  buildPythonPackage,
+  setuptools,
+  flit-core,
+  nltk,
+}:
+
+buildPythonPackage rec {
+  pname = "textblob";
+  version = "0.18.0.post0";
+  format = "pyproject";
```
Comment (dotlambda) [inline pkgs/development/python-modules/textblob/default.nix:13]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/textblob/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-gTHFLGMLzfYdBMNZ+TnJjVuDagH7oiTZ564i/CdODMs=";
+  };
+
+  propagatedBuildInputs = [
+    setuptools
+    flit-core
+    nltk
+  ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/textblob/default.nix:24]:
```suggestion
  build-system = [ flit-core ];

  dependencies = [ nltk ];
```

File: pkgs/development/python-modules/textblob/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  doCheck = false;
```
Comment (dotlambda) [inline pkgs/development/python-modules/textblob/default.nix:26]: Why?

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -9141,6 +9141,8 @@ self: super: with self; {
 
   nltk = callPackage ../development/python-modules/nltk { };
 
+  textblob = callPackage ../development/python-modules/textblob { };
```
Comment (dotlambda) [inline pkgs/top-level/python-packages.nix:9144]: This file should be sorted alphabetically.

**Review by dotlambda — COMMENTED** [review]
`meta.changelog` is missing

File: pkgs/development/python-modules/textblob/default.nix (hunk 3 above)
Comment (idlip) [reply pkgs/development/python-modules/textblob/default.nix]: Changed it to true.

File: pkgs/top-level/python-packages.nix (hunk 4 above)
Comment (idlip) [reply pkgs/top-level/python-packages.nix]: There are some package not in order. I have added textblob in right place.

**Comment by idlip** [comment]
Hey @dotlambda!

Thank you for taking time and reviewing this.

I have updated the changes as suggested!

**Comment by FliegendeWurst** [comment]
@ofborg build python312Packages.textblob

File: pkgs/development/python-modules/textblob/default.nix [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-61B7Yr8ig6cfVr7T4PxO7H04jvdrA2mc+ZQWZXKo2vM=";
+  };
+
+  build-system = [ flit-core ];
+
+  dependencies = [ nltk ];
+
+  doCheck = true;
```
Comment (dotlambda) [inline pkgs/development/python-modules/textblob/default.nix:24]:
```suggestion
```
That's the default. Please specify `nativeCheckInputs`.

File: pkgs/development/python-modules/textblob/default.nix [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+  pythonImportsCheck = [ "textblob" ];
+
+  meta = with lib; {
```
Comment (dotlambda) [inline pkgs/development/python-modules/textblob/default.nix:28]:
```suggestion
meta = {
```
https://github.com/NixOS/nixpkgs/issues/208242

Comment (idlip) [reply pkgs/development/python-modules/textblob/default.nix:28]: I have made the changes, should I make new pr for it? or is there other way?

Comment (FliegendeWurst) [reply pkgs/development/python-modules/textblob/default.nix:28]: Yes, you need to make a new PR

**Pull request merged**